service/controller/resource/updateinfrarefs/create.go

```diff
@@ -22,12 +21,12 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	if err != nil {
 		return microerror.Mask(err)
 	}
-	cc, err := controllercontext.FromContext(ctx)
+	componentVersions, err := r.releaseVersion.ComponentVersion(ctx, cr)
 	if err != nil {
 		return microerror.Mask(err)
 	}
 
-	if len(cc.Status.Versions) == 0 {
+	if len(componentVersions) == 0 {
 		r.logger.LogCtx(ctx, "level", "debug", "message", "no release versions in controller context yet")
```

Comment:
The thing is these checks can be dropped now because we do not rely on the controller context magic anymore.